testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -0,0 +1,480 @@
+        echo "Waiting for pod/validator-{validator_index} IP Address";
+      done;
+      echo -n "${{CFG_SEED_PEER_IP}}" > /opt/libra/data/seed_peer_ip
+  containers:
+  - name: main
+    image: 853397791086.dkr.ecr.us-west-2.amazonaws.com/libra_validator:{image_tag}
+    imagePullPolicy: Always
+    resources:
+      requests:
+        cpu: 7800m
+    ports:
+    - containerPort: 6180
+    - containerPort: 6181
+    - containerPort: 8000
+    - containerPort: 9101
+    - containerPort: 6191
+    volumeMounts:
+    - mountPath: /opt/libra/data
+      name: data
+    env:
+    - name: CFG_NUM_VALIDATORS
+      value: "{num_validators}"
+    - name: CFG_NUM_FULLNODES
+      value: "{num_fullnodes}"
+    - name: CFG_FULLNODE_INDEX
+      value: "{fullnode_index}"
+    - name: CFG_SEED
+      value: "1337133713371337133713371337133713371337133713371337133713371337"
+    - name: CFG_FULLNODE_SEED
+      value: "2674267426742674267426742674267426742674267426742674267426742674"
+    - name: RUST_LOG
+      value: "debug"
+    - name: RUST_BACKTRACE
+      value: "1"
+    - name: CFG_OVERRIDES
+      value: "{cfg_overrides}"
+    - name: MY_POD_IP
+      valueFrom:
+        fieldRef:
+          fieldPath: status.podIP
+    command:
+      - "bash"
+      - "-c"
+      - |
+        set -x;
+        export CFG_LISTEN_ADDR=$MY_POD_IP;
+        export CFG_SEED_PEER_IP=$(cat /opt/libra/data/seed_peer_ip);
+        exec bash /docker-run-dynamic-fullnode.sh
+  volumes:
+  - name: data
+    hostPath:
+      path: /data
+      type: Directory
+  affinity:
+    podAntiAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+          - key: libra-node
+            operator: Exists
+        topologyKey: "kubernetes.io/hostname"
+  terminationGracePeriodSeconds: 5
+  tolerations:
+  - key: "validators"
+    operator: "Exists"
+    effect: "NoSchedule"
+"#,
+            fullnode_index = fullnode_index,
+            num_fullnodes = num_fullnodes,
+            validator_index = validator_index,
+            num_validators = num_validators,
+            node_name = node_name,
+            image_tag = image_tag,
+            cfg_overrides = "",
+            delete_data = delete_data,
+        );
+        let pod_spec: serde_yaml::Value = serde_yaml::from_str(&pod_yaml).unwrap();
+        serde_json::to_vec(&pod_spec).unwrap()
+    }
+
+    async fn get_pod_node_and_ip(&self, pod_name: &str) -> Result<(String, String)> {
+        retry::retry_async(retry::fixed_retry_strategy(5000, 60), || {
+            let pod_api = Api::v1Pod(self.client.clone()).within(DEFAULT_NAMESPACE);
+            let pod_name = pod_name.to_string();
+            Box::pin(async move {
+                match pod_api.get(&pod_name).await {
+                    Ok(o) => {
+                        let node_name = o.spec.node_name.unwrap_or_default();
```

Comment:
This seem a bit cumbersome, if `node_name` is Option, you could just use `ok_or(..)?`

Reply:
Updated